lib/common/client/middleware/logging.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+module Common
+  module Client
+    module Middleware
+      class Logging < Faraday::Middleware
+        DEFAULT_TYPE_KEY = 'undefined'
+
+        def initialize(app, type_key)
+          super(app)
+          @type_key = type_key || DEFAULT_TYPE_KEY
+        end
+
+        def call(env)
+          request_body = Base64.encode64(env.body) if env.body
+
+          @app.call(env).on_complete do |response_env|
+            PersonalInformationLog.create(
```

Comment:
I made a decision to use create vs create! to prevent a failure here from causing the request to fail; particularly since we're not necessarily taking immediate action on all errors reported in sentry. 

Maybe it would be better to log the errors if create is false?

Reply:
I'll pull the bang out, if you feel strongly about logging it let me know and I can add that

Reply:
Nope, looks good to me. It'll be pretty evident if numbers of requests don't match the record count, and not a critical condition.